src/Dofs/DofHandler.jl

```diff
@@ -0,0 +1,250 @@
+export DofHandler, push!, close!, ndofs, ndofs_per_cell, celldofs!, create_sparsity_pattern
+
+"""
+    DofHandler
+
+A `DofHandler` takes care of the degrees of freedom in the system.
+
+```jldoctest dh
+julia> g = generate_grid(Tetrahedron, (2,2,2));
+
+julia> dh = DofHandler(g)
+```
+
+We can now add fields of a certain name and dimension to the dofhandler:
+
+```jldoctest dh
+julia> push!(dh, :temperature, 1);
+
+julia> push!(dh, :displacements, 3);
+```
+
+When we are done with adding the fields, we need to close it:
+
+```jldoctest dh
+julia> close!(dh)
+DofHandler
+  Fields:
+    temperature dim: 1
+    displacements dim: 3
+  Total dofs: 108
+  Dofs per cell: 16
+```
+
+Can now do queries:
+
+```jldoctest dh
+julia> ndofs(dh)
+108
+
+julia> ndofs_per_cell(dh)
+16
+
+julia> global_dofs = zeros(Int, ndofs_per_cell(dh))
+
+julia> dofs_cell!(global_dofs, dh, 3);
+
+julia> global_dofs
+16-element Array{Int64,1}:
+ 13
+ 37
+ 53
+  6
+  7
+  8
+ 14
+ 15
+ 16
+ 38
+ 39
+ 40
+ 54
+ 55
+ 56
+```
+
+Can use it to export
+
+```jldoctest dh
+julia> a = rand(ndofs(dh))
+
+julia> vtkfile = vtk_grid(grid)
+
+julia> vtk_point_data(vtkfile
+```
+
+"""
+
+# TODO: Make this immutable?
+type DofHandler{dim, N, T, M}
+    dofs_nodes::Matrix{Int}
```

Comment:
I guess this is the reason this is not immutable?
We could just make this Vector so it can be resized? IIRC they did this in fenics, with a corresponding offset vector. That can also handle different elements, which this cant